### DISCUSSION TITLE
Creating "substream"s of a StreamProvider

### DISCUSSION THREAD

**Discussion by showier-drastic** [rrousselGit/riverpod#3707] [Q&A]
For example, I have a simple `StreamProvider` that produces random numbers each second:

```dart
Random _random = Random();
final StreamProvider<int> numbersProvider = StreamProvider((ref) async* {
  // keep producing random numbers every second
  while (true) {
    await Future.delayed(const Duration(seconds: 1));
    yield _random.nextInt(100);
  }
});
```

Now I want to create two `StreamProvider`s to be substreams of even and odd numbers:

```dart
// even number provider
final StreamProvider<int> evenNumbersProvider = StreamProvider((ref) async* {
  final value = await ref.watch(numbersProvider.future);
  if (value % 2 == 0) {
    yield value;
  }
});

// odd number provider
final StreamProvider<int> oddNumbersProvider = StreamProvider((ref) async* {
  final value = await ref.watch(numbersProvider.future);
  if (value % 2 != 0) {
    yield value;
  }
});
```

Now the problem is that, at any given time, one of the providers will be in a loading state. For example, if the last generated number is even, then `oddNumbersProvider` will be loading state.

So, could anyone give a hint how can I create two "sub" providers which are never in "loading" state once initial value is populated? Or, to be more general, what's the correct way to "split" one StreamProvider into two or more?

**Comment by snapsl** [comment]
You can use `skipLoadingOnReload=true` on [`AsyncValue`](https://pub.dev/documentation/riverpod/latest/riverpod/AsyncValueX/when.html).
